src/smoosh/src/smoosh_channel.erl

```diff
@@ -65,60 +80,84 @@ close(ServerRef) ->
 flush(ServerRef) ->
     gen_server:call(ServerRef, flush).
 
+is_key(ServerRef, Key) ->
+    gen_server:call(ServerRef, {is_key, Key}).
+
+is_activated(ServerRef) ->
+    gen_server:call(ServerRef, is_activated).
+
+persist(ServerRef) ->
+    gen_server:call(ServerRef, persist).
+
 % gen_server functions.
 
 init(Name) ->
-    schedule_unpause(),
     erlang:send_after(60 * 1000, self(), check_window),
-    {ok, #state{name = Name}}.
+    process_flag(trap_exit, true),
+    Waiting = smoosh_priority_queue:new(Name),
+    State = #state{name = Name, waiting = Waiting, paused = true, activated = false},
+    ok = gen_server:cast(self(), init),
+    {ok, State}.
 
-handle_call({last_updated, Object}, _From, State0) ->
-    {ok, State} = code_change(nil, State0, nil),
+handle_call({last_updated, Object}, _From, State) ->
     LastUpdated = smoosh_priority_queue:last_updated(Object, State#state.waiting),
     {reply, LastUpdated, State};
-handle_call(suspend, _From, State0) ->
-    {ok, State} = code_change(nil, State0, nil),
+handle_call(suspend, _From, State) ->
     #state{active = Active} = State,
     [
         catch erlang:suspend_process(Pid, [unless_suspending])
      || {_, Pid} <- Active
     ],
     {reply, ok, State#state{paused = true}};
-handle_call(resume, _From, State0) ->
-    {ok, State} = code_change(nil, State0, nil),
+handle_call(resume, _From, State) ->
     #state{active = Active} = State,
     [catch erlang:resume_process(Pid) || {_, Pid} <- Active],
-    {reply, ok, State#state{paused = false}};
-handle_call(status, _From, State0) ->
-    {ok, State} = code_change(nil, State0, nil),
+    {reply, ok, State#state{paused = false, activated = true}};
```

Comment:
Why do we set `activated = true` here?

Reply:
to make manual call to resume automatically activate? In this case we should do it on line 62

```
resume(ServerRef) ->
    gen_server:call(ServerRef, activate).
    gen_server:call(ServerRef, resume).
```

OR better

```
resume(ServerRef) ->
    gen_server:call(ServerRef, resume_and_activate).
```

Reply:
Changed `resume` to `resume_and_activate`:
1. https://github.com/apache/couchdb/pull/3766/commits/5c737f17e96d9a331fef2d6cfa8964b25ee8c53a#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R71